### PR TITLE
refactor(html): use state not href

### LINKED ISSUE #1223: (bug) Location Management buttons broken with Angular 1.6.x
The location management buttons are now broken to navigate to villages, sectors, provinces, and countries.  This is likely due to a hardcoded `#/url` and the routing changes with Angular 1.6.x.  I recommend someone fix this with `ui-sref` state routing instead of using hrefs or other routes.

### LINKED ISSUE #1279: (bug) Remove href links from application
A recent PR updated many of angular dependencies including `ui-router`, this has broken many of our older direct `href` links found throughout the application. These should all be removed to ensure that active users to not come across broken links. 

Output: 
```sh
cd bhima/client/src/partials/
grep -rl "ng-href" .

./patients/documents/views/documentView.html
./patients/documents/documents.html
./settings/settings.html
./templates/bhRendererDropdown.tmpl.html
./templates/bhCurrencySelect.tmpl.html
./templates/bhFindDocument.tmpl.html
./templates/breadcrumb.tmpl.html

grep -rl "href=\"" -r .

./fiscal/fiscal.list.html
./patients/edit/edit.html
./patients/documents/views/documentView.html
./patients/documents/documents.html
./locations/locations.html
./locations/province/province.html
./locations/village/village.html
./locations/country/country.html
./locations/sector/sector.html
./patient_invoice/templates/grid/actions.tmpl.html
./profit_center/profit_center.html
./admin/transaction_type/transaction_type.js
./cash/cashboxes/cashboxes.tmp.html
./settings/settings.html
./templates/bhRendererDropdown.tmpl.html
./templates/bhCurrencySelect.tmpl.html
./templates/actionsDropdown.html
./templates/inventoryItems.tmpl.html
./templates/grid/voucherType.tmpl.html
./templates/grid/patient.card.cell.html
./templates/bhFindDocument.tmpl.html
./templates/bhLocationSelect.tmpl.html
./templates/typeahead/accounts.html
./templates/breadcrumb.tmpl.html
./purchases/create/templates/actions.tmpl.html
```

### PR DESCRIPTION
**Pull request by jniles**
This commit refactors a ton of client modules to use `ui-sref` tags
which navigate to particular states instead of `href`s which are
dependent on our URL syntax.

Additionally, it deletes two unused files (profit centers and an
accounts template).

It also updates several HTML files to use the translate directive
instead of the translate filter.

Closes #1279.  Closes #1223.

----
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: client/src/partials/locations/locations.html [hunk 1]
```diff
@@ -9,32 +9,30 @@
 
 <!-- Menu to manage the various locations -->
 <div class="flex-util">
-  <a href="#/locations/village/" id="village" class="btn btn-default btn-sm" style="margin: 0 2px;">
-    <span class="glyphicon glyphicon-home"></span> {{ 'LOCATION.MANAGE_VILLAGE' | translate }}
+  <a ui-sref="locationsVillage" id="village" class="btn btn-default btn-sm" style="margin: 0 2px;" href>
```
Comment (sfount) [inline client/src/partials/locations/locations.html]: Is there a reason these `href`s need to remain? I know that for links the mouse interaction behaviour is not the same without an href element but I'm not aware of this for buttons?

File: client/src/partials/locations/locations.routes.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+angular.module('bhima.routes')
+  .config(['$stateProvider', function ($stateProvider) {
+    $stateProvider
+
+      .state('locations', {
```
Comment (sfount) [inline client/src/partials/locations/locations.routes.js:5]: 👍

**Review by sfount — APPROVED** [review]
Code LGTM. Very thorough!

File: client/src/partials/locations/locations.html (hunk 1 above)
Comment (jniles) [reply client/src/partials/locations/locations.html]: Ah, didn't really pay attention to the class.  I'll make the change.  Good catch.

**Comment by jniles** [comment]
@sfount, updates made.

@lomamech, when you get a chance, can you pull this PR and help @sfount and I test the routes for a few minutes?  I'm not confident our end to end tests cover everything.

**Comment by sfount** [comment]
Updates look good, this can merged at any time although it looks like there's minor conflict in `vouchers/index.js`

**Comment by jniles** [comment]
@sfount, I'll take a look.

**Comment by jniles** [comment]
Tests pass!  Merging.

**Pull request merged**